### PR TITLE
Configure isort to use parentheses instead of line-continuation character

### LINKED ISSUE #13714: `isort` parameter to use parenthesis
There is an `isort` parameter to use parenthesis instead of a line-continuation character: ``--up`` or ``--use-parentheses``.  I agree that we should use that.

__Originally posted by @larrybradley in https://github.com/astropy/astropy/pull/13710#discussion_r974472953__

### PR DESCRIPTION
**Pull request by larrybradley**
### Description

This PR adds the `--use-parentheses` `isort` option in the pre-commit setup.

Fixes #13714

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by adrn — APPROVED** [review]
Thanks @larrybradley - pinging @pllim and @nstarman here too

**Comment by pllim** [comment]
Maybe not... lol https://results.pre-commit.ci/run/github/2081289/1663606867.-IPDiJ3SRI6RcojHVOhaHA

**Comment by nstarman** [comment]
Oh @larrybradley can you squash the commit and run `pre-commit run -a` from your Astropy folder?

**Comment by larrybradley** [comment]
Done!  There were a bunch of `isort` changes (not just parentheses).

**Comment by pllim** [comment]
`+1,270 −1,516` 😱

**Comment by pllim** [comment]
Maybe we went a bit too far with this... lol. The convolution stuff is already a different PR. And I thought we agreed to leave extern alone?

**Review by pllim — CHANGES_REQUESTED** [review]
Blocking merge until the diff is more manageable.

**Comment by larrybradley** [comment]
All I did was run `pre-commit run -a`.  :smile:

**Comment by nstarman** [comment]
> And I thought we agreed to leave extern alone?

Yeah, it shouldn't be able to touch extern! That's in the configuration file.

File: .pre-commit-config.yaml
Comment (WilliamJamieson) [inline .pre-commit-config.yaml:21]:
```suggestion
        args: [--use-parentheses]
```

Why not add `use_parentheses = true` to the `isort` config in the `pyproject.toml` file instead? This would apply to using both `pre-commit` and `isort`. This change only applies when using `pre-commit`

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
Something has gone very wrong in this PR, somehow there are `isort` changes in files which `isort` is configured to ignore: https://github.com/astropy/astropy/blob/e81b367055e5cf5aafbcdfcc2662f50ff257847a/pyproject.toml#L40-L57

The application of `isort` needs to be redone.

**Comment by larrybradley** [comment]
Apparently I ran `pre-commit run -a` from one directory too high.  It's fixed now, and there was only the one line continuation (in stats).  I don't know why the CI failed last time.

I also used @WilliamJamieson 's suggestion to put this configuration in `pyproject.toml`.

**Review by WilliamJamieson — APPROVED** [review]

**Review by nstarman — APPROVED** [review]

**Review by pllim — APPROVED** [review]
LGTM now. Thanks!

**Comment by pllim** [comment]
You might have broken the record of number of approvals for a single PR. 😆  Thanks!

**Pull request merged**
